src/octoprint/plugins/virtual_printer/__init__.py

```diff
@@ -270,24 +309,102 @@ def tcp_command(host, port):
-            @click.argument("path", type=click.Path)
+            @click.option(
+                "--path",
+                "path",
+                type=click.Path(),
+                default="/tmp/octoprint-virtual_printer",
+            )
             def uds_command(path):
+                if os.path.exists(path):
+                    fd = os.open(path, os.O_PATH)
+                    mode = os.fstat(fd).st_mode
+                    os.close(fd)
+                    if stat.S_ISSOCK(mode):
+                        os.unlink(path)
+                    else:
+                        print(
+                            f"{path} already exists and is not a socket. Unable to continue."
+                        )
+                        sys.exit(1)
+
                 click.echo(f"Creating Unix Domain Socket server on {path}...")
+
                 with socket.socket(family=socket.AF_UNIX, type=socket.SOCK_STREAM) as s:
                     s.bind(path)
                     s.listen()
                     while True:
-                        conn, addr = s.accept()
-                        print(f"Connection from {addr}")
+                        conn, _ = s.accept()
+                        print("Connection opened")
                         with conn:
                             virtual = SocketVirtualPrinterWrapper(
                                 conn, settings, datafolder
                             )
                             virtual.wait()
-                        print(f"Connection closed from {addr}")
+                        print("Connection closed")
 
             commands.append(uds_command)
 
+        ##~~ Unix PTY
+
+        @click.command("pty")
+        @click.option(
+            "--path", "path", type=click.Path(), default="/tmp/octoprint-virtual_printer"
+        )
+        def pty_command(path):
+            if os.path.islink(path):
+                realpath = os.path.realpath(path)
+                if not os.path.exists(realpath):
+                    os.unlink(path)
+                else:
+                    print(
+                        f"{path} already exists as a valid symlink. Unable to continue."
+                    )
+                    sys.exit(1)
+            elif os.path.exists(path):
+                print(f"{path} already exists. Unable to continue.")
+                sys.exit(1)
+
+            click.echo(f"Creating Unix PTY on {path}...")
+
+            mfd, sfd = pty.openpty()
+            sfd_path = os.ttyname(sfd)
+            os.symlink(sfd_path, path)
+            fcntl.fcntl(
+                mfd, fcntl.F_SETFL, fcntl.fcntl(mfd, fcntl.F_GETFL) | os.O_NONBLOCK
+            )
+            attrs = termios.tcgetattr(mfd)
+            attrs[3] = attrs[3] & ~termios.ECHO
+            termios.tcsetattr(mfd, termios.TCSADRAIN, attrs)
+
+            virtual = PTYVirtualPrinterWrapper(mfd, settings, datafolder)
+            virtual.wait()
+
+        commands.append(pty_command)
```

Comment:
Needs to go into a wrapper so we don't try to add this on unsupported platforms.

Something like this should work:

``` python
try:
    import fcntl
    import pty
    import termios

    # further code here
except ImportError:
    # unsupported on this platform
```

Reply:
Done